plugins/lightspeed-backend/src/service/router.ts

```diff
@@ -130,11 +130,30 @@ export async function createRouter(
           new MessagesPlaceholder('messages'),
         ]);
 
+        let history: BaseMessage[] = [];
+        try {
+          history = await loadHistory(conversation_id, DEFAULT_HISTORY_LENGTH);
+        } catch (error) {
+          logger.error(`${error}`);
+        }
+
+        const userHistory = history?.map(message =>
+          message?.response_metadata?.role == Roles.HumanRole
+            ? new HumanMessage({ content: message.content })
+            : new AIMessage({ content: message.content }),
+        );
+
         const chain = prompt.pipe(openAIApi);
         let content = '';
+        const userMessageTimestamp = Date.now();
+        let botMessageTimestamp;
         for await (const chunk of await chain.stream({
-          messages: [new HumanMessage(query)],
+          messages: [...userHistory, new HumanMessage(query)],
```

Comment:
since this PR feeds back the previous user history. we might want to introduce a couple new test cases to test the memory of previous conversations.

Reply:
added test cases.